File: backend/voting-app/src/main/java/com/tu/votingapp/entities/surveys/SurveyEntity.java
package com.tu.votingapp.entities.surveys;

import com.tu.votingapp.entities.UserEntity;
import com.tu.votingapp.enums.SurveyStatus;
import jakarta.persistence.*;
import lombok.*;

import java.sql.Date;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "surveys")
@Data
@EqualsAndHashCode(callSuper = false)
@NoArgsConstructor
@AllArgsConstructor
public class SurveyEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false, updatable = false)
    private Long id;

    @Column(name = "survey_title")
    private String title;

    @Column(name = "survey_description", columnDefinition = "TEXT")
    private String description;

    @Column(name = "start_date")
    private Date startDate;

    @Column(name = "end_date")
    private Date endDate;

    @Enumerated(EnumType.STRING)
    @Column(name = "survey_status")
    private SurveyStatus status;

    @OneToMany(mappedBy = "survey", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<SurveyQuestionsEntity> questions = new ArrayList<>();

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "created_by")
    private UserEntity createdBy;
}

File: frontend/android/app/src/main/java/com/example/android/entity/survey/SurveyEntity.kt
package com.example.android.entity.survey

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "surveys")
data class SurveyEntity(
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "id")
    val id: Long? = null,

    @ColumnInfo(name = "survey_title")
    val title: String,

    @ColumnInfo(name = "survey_description")
    val description: String? = null,

    @ColumnInfo(name = "start_date")
    val startDate: String,

    @ColumnInfo(name = "end_date")
    val endDate: String,
    @ColumnInfo(name = "survey_status")
    val status: String,
    @ColumnInfo(name = "created_by")
    val createdById: Long
)